docs/development/release-checklist.md

```diff
@@ -0,0 +1,29 @@
+## Release Checklist
+
+- [ ] Update version number in package.json if not done after last release.
+- [ ] Tag release branch. Wait for the CI to build artifacts.
+- [ ] Sign windows installer.
+- [ ] Sign Linux installer.
```

Comment:
We don't need to sign Linux anymore. This is automated by OBS.

Reply:
Removed `Linux` statement.